src/internal/NeoEspRmtMethod.h

```diff
@@ -0,0 +1,293 @@
+    {
+        // TODO: use "rmt_tx_end_callback" to set channel BUSY->IDLE state (?)
+        updateState();
+        return (_rmtState == NeoRmtState_Idle);
+    }
+
+    void Initialize()
+    {
+        ESP_ERROR_CHECK(get_channel_free(&_channel));
+
+        Serial.printf("[NeoPixelBus] Initializing RMT on channel:%u, pin:%u\n", _channel, _pin);
+
+        rmt_config_t config;
+        config.rmt_mode = RMT_MODE_TX;
+        config.channel = (rmt_channel_t)_channel;
+        config.gpio_num = (gpio_num_t)_pin;
+        config.mem_block_num = 1;
+        config.clk_div = CLK_DIV_RMT;
+        config.tx_config.loop_en = false;
+        config.tx_config.idle_output_en = true;
+        config.tx_config.idle_level = RMT_IDLE_LEVEL_LOW;
+        config.tx_config.carrier_en = false;
+        config.tx_config.carrier_level = RMT_CARRIER_LEVEL_LOW;
+
+        ESP_ERROR_CHECK(rmt_config(&config));
+        ESP_ERROR_CHECK(rmt_driver_install(config.channel, 0, 0));
+        ESP_ERROR_CHECK(rmt_translator_init(config.channel, u8_to_rmt));
+
+        _rmtState = NeoRmtState_Idle;
+    }
+
+    void ICACHE_RAM_ATTR Update()
+    {
+      // wait for not actively sending data
+        while (!IsReadyToUpdate())
+        {
+            yield();
+        }
+
+        FillBuffers();
+        ESP_ERROR_CHECK(rmt_write_sample(_channel, _rmtBuffer, _bufferSize, false));
+
+        _rmtState = NeoRmtState_Sending;
+    }
+
+    uint8_t* getPixels() const
+    {
+        return _pixels;
+    }
+
+    size_t getPixelsSize() const
+    {
+        return _pixelsSize;
+    }
+
+private:
+    uint8_t* _pixels;       // holds LED color values
+    uint8_t* _rmtBuffer;    // holds a copy of '_pixels' (+1 for RESET), cloned on Update()
+
+    size_t  _pixelsSize;    // size of '_pixels' array
+    size_t  _bufferSize;    // size of '_rmtBuffer' array
+    uint8_t _pin;           // output pin number
+
+    rmt_channel_t _channel; // RMT channel
+
+    static const uint32_t _bit0 = (NS_TO_CYCLES(T_SPEED::T0H) | (1 << 15) | (NS_TO_CYCLES(T_SPEED::T0L) << 16));  // logical 0
+    static const uint32_t _bit1 = (NS_TO_CYCLES(T_SPEED::T1H) | (1 << 15) | (NS_TO_CYCLES(T_SPEED::T1L) << 16));  // logical 1
+    static const uint32_t  _rst = (NS_TO_CYCLES(T_SPEED::ResetTimeUs*1000L));                                     // reset
+
+    volatile NeoRmtState _rmtState;
+
+
+    // translate routine (called from RMT ISR) -> sample_to_rmt_fn
+    static void ICACHE_RAM_ATTR u8_to_rmt(const void* src, rmt_item32_t* dest, size_t src_size, size_t wanted_num, size_t* translated_size, size_t* item_num)
+    {
+        if(src == NULL || dest == NULL) {
+            *translated_size = 0;
+            *item_num = 0;
+            return;
+        }
+
+        size_t size = 0;
+        size_t num = 0;
+        uint8_t mask;
+
+        uint8_t *psrc = (uint8_t *)src;
+        rmt_item32_t* pdest = dest;
+
+        bool last = !(src_size > (wanted_num >> 3));
+        if(last) src_size--;
```

Comment:
(Code Style)

* spaces are your friend.  'if' is not a function call, so it should have a space between it and the '()'.  Function calls have no spaces between them.
* One statement per line, if and its branching code should be on separate lines.
* All branching statements must have "{}".  Even though they are optional for single line branches they are required for code style guide.